internal/relation/endpoint_test.go

```diff
@@ -0,0 +1,79 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package relation
+
+import (
+	jc "github.com/juju/testing/checkers"
+	gc "gopkg.in/check.v1"
+
+	corerelation "github.com/juju/juju/core/relation"
+)
+
+type endpointSuite struct{}
+
+var _ = gc.Suite(&endpointSuite{})
+
+func (s *endpointSuite) TestParseRelationKey(c *gc.C) {
+	tests := []struct {
+		relationKey         string
+		endpointIdentifiers []EndpointIdentifier
+	}{{
+		relationKey: "app1:end1 app2:end2",
+		endpointIdentifiers: []EndpointIdentifier{
+			{
+				ApplicationName: "app1",
+				EndpointName:    "end1",
+			}, {
+				ApplicationName: "app2",
+				EndpointName:    "end2",
+			},
+		},
+	}, {
+		relationKey: "app1:end1",
+		endpointIdentifiers: []EndpointIdentifier{
+			{
+				ApplicationName: "app1",
+				EndpointName:    "end1",
+			},
+		},
+	}, {
+		relationKey: "app_1:end_1              app_2:end_2",
+		endpointIdentifiers: []EndpointIdentifier{
+			{
+				ApplicationName: "app_1",
+				EndpointName:    "end_1",
+			}, {
+				ApplicationName: "app_2",
+				EndpointName:    "end_2",
+			},
+		},
+	}}
+
+	for _, test := range tests {
+		result, err := ParseRelationKey(corerelation.Key(test.relationKey))
+		c.Check(err, jc.ErrorIsNil, gc.Commentf("failed on key: %s", test.relationKey))
+		c.Check(result, gc.DeepEquals, test.endpointIdentifiers)
+	}
+}
+
+func (s *endpointSuite) TestParseRelationKeyError(c *gc.C) {
+	tests := []struct {
+		relationKey string
+		errorRegex  string
+	}{{
+		relationKey: "app1:end1 app2:end2 app3:end3",
+		errorRegex:  "expected 1 or 2 endpoints in relation key, found 3.*",
+	}, {
+		relationKey: "",
+		errorRegex:  "expected 1 or 2 endpoints in relation key, found 0.*",
+	}, {
+		relationKey: "app_1:end_1app2:end2",
+		errorRegex:  "expected endpoints of form <application-name>:<endpoint-name>, got.*",
+	}}
+
+	for _, test := range tests {
```

Comment:
Each struct should have a description to be printed out, as well as an index
Test 3 of 5: successful peer relation